Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/DriverFee.hs

```diff
@@ -0,0 +1,36 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Types.DriverFee where
+
+import Data.Time (UTCTime)
+import Domain.Types.Person (Driver)
+import EulerHS.Prelude (Generic, Read, Show)
+import Kernel.Types.Common (Money)
+import Kernel.Types.Id
+
+data DriverFee = DriverFee
+  { id :: Id DriverFee,
+    driverId :: Id Driver,
+    totalAmount :: Money,
+    payBy :: UTCTime,
+    startTime :: UTCTime,
+    endTime :: UTCTime,
+    status :: DriverFeeStatus,
+    createdAt :: UTCTime,
+    updatedAt :: UTCTime
+  }
+  deriving (Generic)
+
+data DriverFeeStatus = ONGOING | PAYMENT_PENDING | PAYMENT_OVERDUE | CLEARED | EXEMPTED deriving (Read, Show)
```

Comment:
And which status means that payment completed by driver? Do you check anyhow status of `PaymentOrder` in your PR?

Reply:
`CLEARED` means the payment is completed. I can add logs if it is not clear with the enums. `PaymentOrder` I am not checking here. Everything related to actual transaction will be done as a separate PR.